Kudu.Core/Infrastructure/PathUtils/PathWindowsUtility.cs

```diff
@@ -108,6 +108,11 @@ internal override string ResolveNpmJsPath()
 
         internal override string ResolveMSBuild15Dir()
         {
+            if (ScmHostingConfigurations.UseLatestMSBuild)
+            {
+                return Path.Combine(ResolveLatestMSBuildDir());
```

Comment:
I would pick and pin to one MSBuild version.   This ensures one deterministic version and, more importantly, no impact/regression if we ship new MSBuild for those scenario.

Reply:
yeah that makes sense. 

How about different scm config for each version something like: UseLatestMSBuildForMSBUILD14 and UseLatestMSBuildforMSBUILD15?

Reply:
Updated the PR.